src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -251,6 +250,7 @@ public IndexShard(ShardId shardId, IndexSettingsService indexSettingsService, In
         logger.debug("state: [CREATED]");
 
         this.checkIndexOnStartup = indexSettings.get("index.shard.check_on_startup", "false");
+        this.translog = newTranslog(shardId, indexSettings, indexSettingsService, bigArrays, indexStore);
```

Comment:
yeah, I hear you. The problem is that some places in the code rely on the translog being there. Ex. constructor IndexShardGateway :
```
 } else if (syncInterval.millis() > 0) {
            this.indexShard.translog().syncOnEachOperation(false);
```

maybe we can do this as another change as well?